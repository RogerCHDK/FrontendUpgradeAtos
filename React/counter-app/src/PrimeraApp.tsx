import React from 'react'
import PropTypes from 'prop-types'


interface Props{
    saludo?: string
}

const PrimeraApp: React.FC<Props> = ( { saludo }) =>{
    return <h1>{ saludo }</h1>
}


PrimeraApp.propTypes = {
    saludo: PropTypes.string
}

PrimeraApp.defaultProps = {
    saludo: 'Hola soy Goku'
  };

export default PrimeraApp;