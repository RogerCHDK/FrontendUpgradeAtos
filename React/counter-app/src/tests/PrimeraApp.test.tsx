// import { render, screen } from "@testing-library/react";
import { shallow } from 'enzyme';
import React from 'react'
import PrimeraApp from '../PrimeraApp';


describe('Probando mi primer component', () => {
    
    // test('should show Hola Mundo in the component', () => {
    //     const saludo = 'Rogelio';

    //     render(<PrimeraApp saludo={saludo}/>);

    //     expect( screen.getByText( saludo ) ).toBeInTheDocument();
    // });

    test('should show the component PrimeraApp succesfully', () => {
        const saludo = 'Hola Rogelio';

        const wrapper = shallow( <PrimeraApp saludo={ saludo }/>);

        expect( wrapper ).toMatchSnapshot();
    });
});