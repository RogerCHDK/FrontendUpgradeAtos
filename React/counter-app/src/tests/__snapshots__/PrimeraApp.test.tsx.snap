// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando mi primer component should show the component PrimeraApp succesfully 1`] = `
<h1>
  Hola Rogelio
</h1>
`;
